types/theo/index.d.ts

```diff
@@ -0,0 +1,98 @@
+// Type definitions for theo 8.0
+// Project: https://github.com/salesforce-ux/theo
+// Definitions by: Pete Petrash <https://github.com/petekp>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import { Iterable, Collection, Map, List, OrderedMap } from "immutable";
+
+export type StyleProperty =
+    | "name"
+    | "value"
+    | "type"
+    | "originalValue"
+    | "category"
+    | "comment"
+    | "meta"
+    | string;
```

Comment:
What is the point of specifying values if all of them are overridden by 'string'?

Reply:
This may not be the right way to handle it, but the reason for having `string` is because users of this package are also able to define their own arbitrary StyleProperty values.

I wonder if it would be better to change this type to represent only the package-provided StyleProperty types, and to create a new type for UserDefinedStyleProperties that could be extended by the user?

Any advice for handling this is welcome!

Reply:
Yes, unfortunately `| string` defeats the purpose of the whole type. The typechecker will collapse this entire type to `string` so users of your typings will never even see the individual literal values. So you could go with just `string` or the literal values, but not both. I think your idea of removing `string` and representing only the package-provided types is good.